drivers/modem/ublox-sara-r4.c

```diff
@@ -660,6 +677,61 @@ static void on_cmd_atcmdinfo_rssi(struct net_buf **buf, u16_t len)
 	ictx.mdm_ctx.data_rssi = -1000;
 }
 
+/* Handler: +CSQ: <signal_power>[0],<qual>[1] */
+static void on_cmd_atcmdinfo_rssi_csq(struct net_buf **buf, u16_t len)
+{
+	int i = 0, rssi, param_count = 0;
+	size_t value_size;
+	char value[12];
+
+	value_size = sizeof(value);
+	while (*buf && len > 0 && param_count < 1) {
+		i = 0;
+		(void)memset(value, 0, value_size);
+
+		while (*buf && len > 0 && i < value_size) {
+			value[i] = net_buf_pull_u8(*buf);
+			len--;
+			if (!(*buf)->len) {
+				*buf = net_buf_frag_del(NULL, *buf);
+			}
+
+			/* "," marks the end of each value */
+			if (value[i] == ',') {
+				value[i] = '\0';
+				break;
+			}
+
+			i++;
+		}
+
+		if (i == value_size) {
+			i = -1;
+			break;
+		}
+
+		param_count++;
+	}
+
+	if (param_count == 1 && i > 0) {
+		rssi = atoi(value);
+		if (rssi == 31) {
+			ictx.mdm_ctx.data_rssi = -46;
+		} else if (rssi >= 0 && rssi <= 31) {
+			/* FIXME: This value depends on the RAT */
+			ictx.mdm_ctx.data_rssi = -70;
```

Comment:
I made the assumption it would most often be 3g therefore used:

		`	ictx.mdm_ctx.data_rssi = -110 + ((rssi * 2) + 1);`